Failing random tests

* [ ] `Previous node dictated placement (prop testing) (with seed=957474021)`
* [ ] ` Non dictated placement (prop testing) (with seed=2110940700)`

The logs for first case:
```
 FAIL  src/components/ComponentBrowser/__tests__/placement.test.ts > Previous node dictated placement (prop testing) (with seed=957474021)
Error: Property failed after 14 tests
{ seed: 957474021, path: "13:8:1:5:2:2:2:4:17:17:17:22:1:1:6:3:1:17:1:7:2:1:17:1:8:17:1:1:17:1:25:9:20:20:18:18:18:18:18:19:29:29:29:29:29:29:29:29:29:17:1:24:1:8:17:1:1:17:1:40:37:37:37:37:37:41:25:9:20:1:1:4:2:3:1:18:1:6:1:18:1:1:1:1:8:1:17:1:8:1:32:17:1:9:17:1:25:9:17:1:29:5:2:1:1:1:17:1:6:1:19:1:1:32:17:1:8:17:1:17:1:29:6:1:18:1:1:1:8:1:17:1:8:1:31:17:1:24:8:17:1:23:1:7:2:17:1:8:1:17:1:38:24:8:17:1:24:1:8:17:1:17:1:38:24:8:24:1:1:38:24:8:31:17:1:31:17:1:47:26:3:1:4:4:3:16:3:3:3:15:27:13:16:13:15:27:13:13:17:13:13:13:27:13:13:17:29:13:14:27:13:27:27:27:37:37:37:37:38:39:37:37:37:38:37:38:37:37:37:37:37:37:37:37:37:37:37:37:37:37:37:1:1:6:14:19:39:39:39:39:39:39:39:39:39:39:39:39:39:39:39:39:40:3:4:3:4:4:4:16:17:16:17:17:17:16:27:27:27:28:30:27:40:3:3:3:7:3:3:3:4:6:6:10:6:6:6:6:8:8:8:8:8:9:10:8:21:3:3:7:3:3:4:7:7:7:11:7:7:9:7:8:8:8:8:8:9:8:22:3:3:3:8:3:3:3:3:6:10:10:7:7:7:7:7:7:13:7:7:8:3:3:8:6:6:6:11:9:7:7:7:12:11:3:9:4:6:12:7:7:9:7:8:3:3:3:11:3:6:6:6:14:6:7:14:9:3:3:12:6:6:15:7:7:8:3:12:6:15:7:7:17:3:15:34:26:12:15:7:17:17:3:4:3:3:3:4:3:3:6:7:6:6:6:7:6:6:8:7:7:7:8:7:7:8", endOnFailure: true }
Counterexample: [{"nodeData":[{"left":124,"top":0,"width":0,"height":25},{"left":0,"top":0,"width":0,"height":0},{"left":149,"top":0,"width":0,"height":25},{"left":0,"top":0,"width":0,"height":0},{"left":0,"top":0,"width":0,"height":0},{"left":0,"top":0,"width":0,"height":25},{"left":0,"top":0,"width":0,"height":0},{"left":0,"top":0,"width":0,"height":0},{"left":0,"top":0,"width":0,"height":0},{"left":0,"top":0,"width":0,"height":0},{"left":0,"top":0,"width":0,"height":0},{"left":0,"top":0,"width":0,"height":0},{"left":0,"top":0,"width":0,"height":0},{"left":0,"top":0,"width":0,"height":0},{"left":0,"top":0,"width":0,"height":0},{"left":0,"top":0,"width":0,"height":0}],"firstSelectedNode":7}]
Shrunk 434 time(s)
Got AssertionError: Please visually inspect the code below at https://play.vuejs.org/:
<script setup>
const rects = [
  { left: 124, top: 0, width: (124 - 124), height: (25 - 0) },
  { left: 0, top: 0, width: (0 - 0), height: (0 - 0) },
  { left: 149, top: 0, width: (149 - 149), height: (25 - 0) },
  { left: 0, top: 0, width: (0 - 0), height: (0 - 0) },
  { left: 0, top: 0, width: (0 - 0), height: (0 - 0) },
  { left: 0, top: 0, width: (0 - 0), height: (25 - 0) },
  { left: 0, top: 0, width: (0 - 0), height: (0 - 0) },
]

const selectedRects = [
  { left: 0, top: 0, width: (0 - 0), height: (0 - 0) },
  { left: 0, top: 0, width: (0 - 0), height: (0 - 0) },
  { left: 0, top: 0, width: (0 - 0), height: (0 - 0) },
  { left: 0, top: 0, width: (0 - 0), height: (0 - 0) },
  { left: 0, top: 0, width: (0 - 0), height: (0 - 0) },
  { left: 0, top: 0, width: (0 - 0), height: (0 - 0) },
  { left: 0, top: 0, width: (0 - 0), height: (0 - 0) },
  { left: 0, top: 0, width: (0 - 0), height: (0 - 0) },
  { left: 0, top: 0, width: (0 - 0), height: (0 - 0) },
]

const rect = { left: 148, top: 24, width: 100, height: 20 }
</script>

<template>
<div style="height: 1000px; width: 2000px;">
<div v-for="rect in rects" :style="{ position: 'absolute', left: rect.left + 'px', top: rect.top + 'px', width: rect.width + 'px', height: rect.height + 'px', background: '#88888822' }"></div>
<div v-for="rect in selectedRects" :style="{ position: 'absolute', left: rect.left + 'px', top: rect.top + 'px', width: rect.width + 'px', height: rect.height + 'px', background: '#4444aa44' }"></div>
<div :style="{ position: 'absolute', left: rect.left + 'px', top: rect.top + 'px', width: rect.width + 'px', height: rect.height + 'px', background: '#88222288' }"></div>
</div>
</template>
: expected true to be false // Object.is equality
    at /home/adam-praca/dev/enso/app/gui2/src/components/ComponentBrowser/__tests__/placement.test.ts:329:18
    at file:///home/adam-praca/dev/enso/node_modules/@fast-check/vitest/lib/esm/internals/TestBuilder.js:28:79
    at file:///home/adam-praca/dev/enso/node_modules/@fast-check/vitest/lib/esm/internals/TestWithPropRunnerBuilder.js:3:41
    at AsyncProperty.predicate (file:///home/adam-praca/dev/enso/node_modules/fast-check/lib/esm/check/property/AsyncProperty.js:14:59)
    at AsyncProperty.run (file:///home/adam-praca/dev/enso/node_modules/fast-check/lib/esm/check/property/AsyncProperty.generic.js:46:39)
    at asyncRunIt (file:///home/adam-praca/dev/enso/node_modules/fast-check/lib/esm/check/runner/Runner.js:33:36)
    at file:///home/adam-praca/dev/enso/node_modules/@fast-check/vitest/lib/esm/internals/TestWithPropRunnerBuilder.js:22:9
    at runTest (file:///home/adam-praca/dev/enso/node_modules/@vitest/runner/dist/index.js:613:11)
    at runSuite (file:///home/adam-praca/dev/enso/node_modules/@vitest/runner/dist/index.js:732:15)
    at runFiles (file:///home/adam-praca/dev/enso/node_modules/@vitest/runner/dist/index.js:783:5)

Hint: Enable verbose mode in order to have the list of all failing values encountered during the run
 ❯ src/components/ComponentBrowser/__tests__/placement.test.ts:329:18
    327|         )
    328|       },
    329|     } as string).toBe(false)
       |                  ^
    330|   }
    331|   expect(getMousePosition, 'Should not depend on `mousePosition`').not.toHaveBeenCalled()
 ❯ ../../node_modules/@fast-check/vitest/lib/esm/internals/TestBuilder.js:28:79
 ❯ ../../node_modules/@fast-check/vitest/lib/esm/internals/TestWithPropRunnerBuilder.js:3:41
 ❯ AsyncProperty.predicate ../../node_modules/fast-check/lib/esm/check/property/AsyncProperty.js:14:59
 ❯ AsyncProperty.run ../../node_modules/fast-check/lib/esm/check/property/AsyncProperty.generic.js:46:39
 ❯ asyncRunIt ../../node_modules/fast-check/lib/esm/check/runner/Runner.js:33:36
 ❯ ../../node_modules/@fast-check/vitest/lib/esm/internals/TestWithPropRunnerBuilder.js:22:9
 ❯ buildError ../../node_modules/fast-check/lib/esm/check/runner/utils/RunDetailsFormatter.js:126:15
 ❯ asyncThrowIfFailed ../../node_modules/fast-check/lib/esm/check/runner/utils/RunDetailsFormatter.js:143:11

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/1]⎯

 Test Files  1 failed | 19 passed (20)
      Tests  1 failed | 407 passed (408)
   Start at  14:57:08
   Duration  8.88s (transform 1.53s, setup 3ms, collect 10.87s, tests 4.85s, environment 20.26s, prepare 5.75s)
```

--- a/app/gui2/src/components/ComponentBrowser/placement.ts
+++ b/app/gui2/src/components/ComponentBrowser/placement.ts
@@ -1,5 +1,4 @@
 import { bail } from '@/util/assert'
-import { MultiRange, Range } from '@/util/range'
 import { Rect } from '@/util/rect'
 import { Vec2 } from '@/util/vec2'
 
@@ -42,23 +41,12 @@ export function nonDictatedPlacement(
   const initialRect = new Rect(initialPosition, nodeSize)
   let top = initialPosition.y
   const height = nodeSize.y
-  const minimumVerticalSpace = height + gap * 2
   const bottom = () => top + height
-  const occupiedYRanges = new MultiRange()
-  for (const rect of nodeRects) {
+  const nodeRectsSorted = Array.from(nodeRects).sort((a, b) => a.top - b.top)
+  for (const rect of nodeRectsSorted) {
     if (initialRect.intersectsX(rect) && rect.bottom + gap > top) {
-      if (rect.top - bottom() >= gap) {
-        const range = new Range(rect.top, rect.bottom)
-        occupiedYRanges.insert(range, range.expand(gap))
-      } else {
+      if (rect.top - bottom() < gap) {
         top = rect.bottom + gap
-        const rangeIncludingTop = occupiedYRanges
-          .remove(new Range(-Infinity, rect.bottom + minimumVerticalSpace))
-          .at(-1)
-        if (rangeIncludingTop) {
-          top = Math.max(top, rangeIncludingTop.end + gap)
-          occupiedYRanges.remove(rangeIncludingTop)
-        }
       }
     }
   }
@@ -105,24 +93,13 @@ export function previousNodeDictatedPlacement(
   let left = initialLeft
   const width = nodeSize.x
   const right = () => left + width
-  const minimumHorizontalSpace = width + gap * 2
   const initialPosition = new Vec2(left, top)
   const initialRect = new Rect(initialPosition, nodeSize)
-  const occupiedXRanges = new MultiRange()
-  for (const rect of nodeRects) {
+  const sortedNodeRects = Array.from(nodeRects).sort((a, b) => a.left - b.left)
+  for (const rect of sortedNodeRects) {
     if (initialRect.intersectsY(rect) && rect.right + gap > left) {
-      if (rect.left - right() >= gap) {
-        const range = new Range(rect.left, rect.right)
-        occupiedXRanges.insert(range, range.expand(gap))
-      } else {
+      if (rect.left - right() < gap) {
         left = rect.right + gap
-        const rangeIncludingLeft = occupiedXRanges
-          .remove(new Range(-Infinity, rect.right + minimumHorizontalSpace))
-          .at(-1)
-        if (rangeIncludingLeft) {
-          left = Math.max(left, rangeIncludingLeft.end + gap)
-          occupiedXRanges.remove(rangeIncludingLeft)
-        }
       }
     }
   }

